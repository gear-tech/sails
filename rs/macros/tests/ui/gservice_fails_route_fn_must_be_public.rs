use sails_macros::{route, service};

struct MyService;

#[service]
impl MyService {
    pub fn do_this(&mut self, p1: u32, p2: String) -> String {
        format!("{p1}: ") + &p2
    }

    #[route("this")]
    fn this(&self, p1: bool) -> bool {
        !p1
    }
}

#[tokio::main]
async fn main() {}
