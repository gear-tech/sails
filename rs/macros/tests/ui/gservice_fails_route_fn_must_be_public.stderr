error: `route` attribute can be applied to public methods only
  --> tests/ui/gservice_fails_route_fn_must_be_public.rs:12:5
   |
12 |     fn this(&self, p1: bool) -> bool {
   |     ^^
